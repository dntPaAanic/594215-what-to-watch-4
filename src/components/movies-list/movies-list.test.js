import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';

const films = [
  {
    title: `The Grand Budapest Hotel`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: `2001`,
  },
  {
    title: `Bohemian Rhapsody`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: `2014`,
  },
  {
    title: `Macbeth`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: `2000`,
  },
  {
    title: `Aviator`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: `2012`,
  },
  {
    title: `We need to talk about Kevin`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Crime`,
    releaseDate: `2018`,
  },
  {
    title: `What We Do in the Shadows`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: `2005`,
  },
  {
    title: `Pulp Fiction`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: `2003`,
  },
  {
    title: `No Country for Old Men`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: `2014`,
  },
];

it(`MoviesList should render correct`, () => {
  const tree = renderer
    .create(<MoviesList films={films} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
