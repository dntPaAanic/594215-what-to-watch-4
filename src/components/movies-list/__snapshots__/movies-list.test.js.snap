// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render correct 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/0"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          The Grand Budapest Hotel
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/1"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          Bohemian Rhapsody
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/2"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          Macbeth
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/3"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          Aviator
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/4"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          We need to talk about Kevin
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/5"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          What We Do in the Shadows
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/6"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          Pulp Fiction
        </span>
      </h3>
    </Link>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Link
      style={
        Object {
          "color": "unset",
        }
      }
      to="/films/7"
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <span
          className="small-movie-card__link"
          href="#"
        >
          No Country for Old Men
        </span>
      </h3>
    </Link>
  </article>,
]
`;
